### PR TITLE
feature/48-refresh-token-in-client-cli

### PR DESCRIPTION
**Pull request by vadkor**

### CODE REVIEW

File: client-cli/pom.xml [hunk 1]
```diff
@@ -46,6 +46,11 @@
             <groupId>org.bouncycastle</groupId>
             <artifactId>bcprov-jdk15on</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.nimbusds</groupId>
+            <artifactId>nimbus-jose-jwt</artifactId>
+            <version>8.2.1</version>
```
Comment (fyang1024) [inline client-cli/pom.xml]: Do we care to define the version in the properties section and reference the property here?

File: client-cli/src/main/java/au/org/consumerdatastandards/client/cli/support/ApiUtil.java [hunk 2]
```diff
@@ -77,9 +103,25 @@ public static ApiClient createApiClient(ApiClientOptions clientOptions) throws A
...
+            Map<String, Object> claims = null;
+            if (StringUtils.isNotBlank(accessToken)) {
+                try {
+                    claims = parseClaims(accessToken);
+                    exp = (Integer) claims.get("exp");
+                } catch (ApiException e) {
+                    LOGGER.info("Invalid access token.");
+                }
+            }
+            if (exp == null || exp.longValue() * 1000 < System.currentTimeMillis() + 10000) {
```
Comment (fyang1024) [inline client-cli/src/main/java/au/org/consumerdatastandards/client/cli/support/ApiUtil.java]: Shall we define a constant for 10000 here and give the constant a name like `EXPIRY_BUFFER`?

File: client-cli/pom.xml (hunk 1 above)
Comment (vadkor) [reply client-cli/pom.xml]: The only time the property is really needed is when we have multiple dependencies of the same group with different artifact IDs but the same version. If we use Nimbus JWT anywhere else we'll just remove the version here and declare it in pom's dependencyManagement section.

File: client-cli/src/main/java/au/org/consumerdatastandards/client/cli/support/ApiUtil.java (hunk 2 above)
Comment (vadkor) [reply client-cli/src/main/java/au/org/consumerdatastandards/client/cli/support/ApiUtil.java]: We shall!

**Pull request merged**